docs/abi.md

```diff
@@ -20,12 +20,39 @@ involve manual management of linear memory. Oak Applications will typically use
 the higher-level [Oak SDK](sdk.md) which provides more convenient (and safer)
 wrappers around this functionality.
 
+## Integer Types
+
+WebAssembly has two integer types (`i32` and `i64`) which are treated as
+[signed or unsigned values depending on the context](https://webassembly.github.io/spec/core/syntax/types.html#value-types).
+
+In this document, integer types are written as `u32` or `u64` to indicate that
+the corresponding values will be treated as unsigned values. (There are
+currently no uses of signed values in the ABI).
```

Comment:
Maybe this should say that we use a more precise description for the types in the API, referring to the interpretation of those values as either signed or unsigned in the target language. I guess this relies on C++, Rust and other languages to perform conversions between those types in a compatible way (which hopefully is the case, but I have not checked). In fact, if you know of any such reference, it would be good to include a link here, so we know what we can safely rely on for each language.

I think my preferred approach would be to keep using i32 / i64 in the API function signatures, and explain how to interpret those values in the description of each method. But I'll leave that to you to decide anyways.